### DISCUSSION TITLE
func.__name__ is empty with --assert-import or --assert-call

### DISCUSSION THREAD

**Discussion by joonis** [dashingsoft/pyarmor#1199] [Ideas]
Obfuscated with `--assert-import` or `--assert-call` the following script prints an empty string.

```
def foo():
    pass

print(foo.__name__)
```

**Comment by jondy** [comment]
This is expected result, these features will hide function name.

**Comment by joonis** [comment]
What is the benefit for this?
By the way you can get the name by the attribute `__qualname__` as well.

**Comment by jondy** [comment]
The design is when any restrict option enable, co_name is clear.

So `__name__` is clear also. If the script need `__name__`, `__qualname__` could be used in most of cases.

**Comment by joonis** [comment]
But what is the advantage when `__name__` is cleared?
It somewhat destroys the output of tracebacks, the `help()` function and the functionality of decorators which depend on the `__name__` attribute.

**Comment by jondy** [comment]
Well, `--assert-call` or any restrict option will not touch `__name__` in 8.2

If users want to clear function name, config it explicitly
```
pyarmor cfg mix_coname = 1
```
